Sources/OpenAPIRuntime/Conversion/Configuration.swift

```diff
@@ -25,14 +25,45 @@ public protocol DateTranscoder: Sendable {
 }
 
 /// A transcoder for dates encoded as an ISO-8601 string (in RFC 3339 format).
-public struct ISO8601DateTranscoder: DateTranscoder {
+public struct ISO8601DateTranscoder: DateTranscoder, @unchecked Sendable {
+
+    /// The lock protecting the formatter.
+    private let lock: NSLock
+
+    /// The underlying date formatter.
+    private let locked_formatter: ISO8601DateFormatter
+
+    /// Creates a new default transcoder.
+    public init() { self.init(optionsOverride: nil) }
+
+    /// Creates a new transcoder with the provided options.
+    /// - Parameter optionsOverride: Options to override the default ones. If you provide
+    ///   nil here, the default options are used.
+    public init(optionsOverride: ISO8601DateFormatter.Options?) {
+        let formatter = ISO8601DateFormatter()
+        if let optionsOverride { formatter.formatOptions = optionsOverride }
+        self.init(formatter: formatter)
+    }
+
+    /// Creates a new transcoder with the provided formatter.
+    /// - Parameter formatter: The formatter to use.
+    public init(formatter: ISO8601DateFormatter) {
```

Comment:
Do we need this one? Technically, the formatter is an open class, so someone could have a subclass they'd want to provide.

Reply:
If we don't need it now, let's drop it until we have a need.

Reply:
It's a spectrum, technically even customizing the options isn't needed, as adopters can implement a whole new date transcoder and have full flexibility. It's just been a common ask.

I think the question of this initializer is kind of related to the locking around the value. Do we create a new formatter under the hood for every call, or do we cache one and protect it with a lock? I haven't measured the performance difference here, maybe I should to inform this discussion more.